whitesource/util.py

```diff
@@ -1,10 +1,150 @@
 import functools
+import json
+
+import tabulate
 
 import ci.util
 import whitesource.client
+import mailutil
 
 
 @functools.lru_cache()
 def create_whitesource_client(whitesource_cfg_name: str):
     cfg_fac = ci.util.ctx().cfg_factory()
     return whitesource.client.WhitesourceClient(cfg_fac.whitesource(whitesource_cfg_name))
+
+
+def generate_reporting_tables(report,
+                              threshold: float,
+                              tablefmt):
+
+    # monkeypatch: disable html escaping
+    tabulate.htmlescape = lambda x: x
+
+    # split respecting CVSS-V3 threshold
+    above = {}
+    below = {}
+
+    for lib, dic in report.items():
+        if float(dic['CVSS-V3']) > threshold:
+            above[lib] = dic
+        else:
+            below[lib] = dic
+
+    def _sort_struct(struct):
```

Comment:
I think `struct` is a rather unconventional name in python; also, it is a special kind of struct (-> a vulnerability-something - yes, good names are hard^^)

additional nit: if you would sort already in line 28, you could save one line of code (and might actually inline the sorting function, thus saving some more boilerplate, and not have to think about a good name) :-P